core/src/main/scala/kafka/server/ClientMetricsManager.scala

```diff
@@ -0,0 +1,85 @@
+package kafka.server
+
+import kafka.Kafka.info
+import kafka.metrics.clientmetrics.{ClientMetricsCache, ClientMetricsConfig, CmClientInformation, CmClientInstanceState}
+import kafka.network.RequestChannel
+import kafka.server.ClientMetricsManager.getSupportedCompressionTypes
+import org.apache.kafka.common.Uuid
+import org.apache.kafka.common.message.GetTelemetrySubscriptionsResponseData
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.record.CompressionType
+import org.apache.kafka.common.requests.{GetTelemetrySubscriptionRequest, GetTelemetrySubscriptionResponse}
+
+import java.util.{Calendar, Properties}
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters._
+
+object ClientMetricsManager {
+  private val _instance = new ClientMetricsManager
+  def getInstance = _instance
+
+  def getSupportedCompressionTypes: List[java.lang.Byte] = {
+    val compressionTypes = new ListBuffer[java.lang.Byte]
+    CompressionType.values.foreach(x => compressionTypes.append(x.id.toByte))
+    compressionTypes.toList
+  }
+
+  def processGetTelemetrySubscriptionRequest(request: RequestChannel.Request,
+                                             throttleMs: Int): GetTelemetrySubscriptionResponse = {
+    val subscriptionRequest = request.body[GetTelemetrySubscriptionRequest]
+    val clientInfo = CmClientInformation(request, subscriptionRequest.getClientInstanceId.toString)
+    _instance.processGetSubscriptionRequest(subscriptionRequest, clientInfo, throttleMs)
+  }
+
+}
+
+class ClientMetricsManager {
+  def getClientInstance(id: Uuid) = ClientMetricsCache.getInstance.get(id)
+
+  def processGetSubscriptionRequest(subscriptionRequest: GetTelemetrySubscriptionRequest,
+                                    clientInfo: CmClientInformation,
+                                    throttleMs: Int): GetTelemetrySubscriptionResponse = {
+    var clientInstanceId = subscriptionRequest.getClientInstanceId
+    if (clientInstanceId == null || clientInstanceId == Uuid.ZERO_UUID) {
+      clientInstanceId = Uuid.randomUuid()
+    }
+    var clientInstance = getClientInstance(clientInstanceId)
+    if (clientInstance == null) {
+      clientInstance = createClientInstance(clientInstanceId, clientInfo)
+    }
+
+    val data =  new GetTelemetrySubscriptionsResponseData()
+        .setThrottleTimeMs(throttleMs)
+        .setClientInstanceId(clientInstanceId)
+        .setSubscriptionId(clientInstance.getSubscriptionId) // TODO: should we use LONG instead of int?
+        .setAcceptedCompressionTypes(getSupportedCompressionTypes.asJava)
+        .setPushIntervalMs(clientInstance.getPushIntervalMs)
+        .setDeltaTemporality(true)
```

Comment:
Maybe reference `KafkaConfig.ClientMetricsDeltaTemporality` instead of the constant of `true`?

Reply:
In fact, I had that change initially but since we have agreed in our meeting that this value is always TRUE on broker side,  I have deleted that and hard coded it.